lib/avromatic/model/attributes.rb

```diff
@@ -21,14 +21,15 @@ def initialize(field)
       end
 
       class AttributeDefinition
-        attr_reader :name, :setter_name, :type, :field, :default, :owner
+        attr_reader :name, :name_string, :setter_name, :type, :field, :default, :owner
         delegate :serialize, to: :type
 
         def initialize(owner:, field:, type:)
           @owner = owner
           @field = field
           @type = type
           @name = field.name.to_sym
+          @name_string = field.name.to_s
```

Comment:
Is `field.name` guaranteed to be a frozen string? If not should we dup and then freeze it? The real memory allocation savings come from the `initialize` method below right?